File: BancoCoreSOAP/src/main/java/com/example/BancoCoreSoap/repository/TransacaoRepository.java
package com.example.BancoCoreSoap.repository;

import com.example.BancoCoreSoap.domain.Transacao;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface TransacaoRepository extends JpaRepository<Transacao, String>{

}

File: BancoRestApi/src/main/kotlin/org/example/bancorestapi/repository/TransacaoRepository.kt
package org.example.bancorestapi.repository

import org.example.bancorestapi.domain.Conta
import org.springframework.data.jpa.repository.JpaRepository
import org.example.bancorestapi.domain.Transacao
import org.springframework.stereotype.Repository

@Repository
interface TransacaoRepository : JpaRepository<Transacao, String> {

    fun findAllByContaNumeroConta(numeroConta: String): List<Transacao>
}